src/app/plugins/plugin_visualize.cpp

```diff
@@ -101,12 +109,27 @@ void PluginVisualize::DrawCameraImage(
     unsigned int n = vis_frame->data.getNumPixels();
     rgb * vis_ptr = vis_frame->data.getPixelData();
     rgb color;
+
     for (unsigned int i = 0; i < n; i++) {
       color = vis_ptr[i];
       color.r = color.g = color.b = ((color.r + color.g + color.b)/3);
       vis_ptr[i] = color;
     }
   }
+
+  if (_v_mask->getBool()) {
+    unsigned int n = vis_frame->data.getNumPixels();
+    rgb * vis_ptr = vis_frame->data.getPixelData();
+    rgb color;
+
+    _image_mask.lock();
+    for (unsigned int i = 0; i < n; i++) {
+      vis_ptr[i].r &= _image_mask.getMask().getPixel(i).getIntensity();
+      vis_ptr[i].g &= _image_mask.getMask().getPixel(i).getIntensity();
+      vis_ptr[i].b &= _image_mask.getMask().getPixel(i).getIntensity();
+    }
+    _image_mask.unlock();
```

Comment:
I imagine the slow down when visualizing is due to this block. We should be able to replace this with the opencv mask functions or at a minimum memset/memcpy. But I think opencv will take advantage of vectorizing loops for certain calls, so we'll probably have better luck with that.

Its not a blocker for merging this, since the core functionality doesn't rely on this, but maybe we should at least add a TODO note for future people looking at this section until it is fixed.

Reply:
In any case, it will add to the overall processing time for visualization, which is already a lot. I think there is no real benefit in having the black background compared to the hull.

I've removed this visualization completely (and kept the hull). If you disagree, let me know and I revert it.